fml/ascii_trie.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#include "flutter/fml/ascii_trie.h"
+
+namespace fml {
+typedef AsciiTrie::TrieNode TrieNode;
+
+namespace {
+void Add(TrieNode** trie, const char* entry) {
+  int ch = entry[0];
+  assert(ch < 128);
+  if (ch != 0) {
+    if (!*trie) {
+      TrieNode* newNode = (TrieNode*)calloc(sizeof(TrieNode), 1);
+      *trie = newNode;
+    }
+    Add(&(*trie)->children[ch], entry + 1);
+  }
+}
+
+TrieNode* MakeTrie(const std::vector<std::string>& entries) {
+  TrieNode* result = nullptr;
+  for (const std::string& entry : entries) {
+    Add(&result, entry.c_str());
+  }
+  return result;
+}
+
+void FreeNode(TrieNode* node) {
+  for (int i = 0; i < 128; ++i) {
+    if (node->children[i]) {
+      TrieNode* child = node->children[i];
+      if (child) {
+        FreeNode(child);
+      }
+    }
+  }
+  free(node);
+}
+}  // namespace
+
+AsciiTrie::~AsciiTrie() {
+  if (node_) {
+    FreeNode(node_);
+  }
+}
+
+void AsciiTrie::fill(const std::vector<std::string>& entries) {
```

Comment:
Here and elsewhere, follow engine naming conventions for naming methods.

Reply:
Can you be more explicit, what do you want this to be named?

Reply:
Ahh, caps!  I forgot we do caps first, shows how long it's been since i contributed to our c++ codebase.  One sec.  It would be helpful if you just said that =P

Reply:
Done.